### PR TITLE
[gff.byn.gff_blin] Initial release

### PR DESCRIPTION
**Pull request by dyacob**
This update provides a cleanup of identical words that had separate entries due to punctuation use (apostrophe vs single left or right curly quote).

### CODE REVIEW

**Comment by keyman-server** [comment]
This pull request is from an external repo and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Comment by darcywong00** [comment]
While you're cleaning up punctuation, I ran your wordlist through unist and saw 21 instances of period `.`. Just checking those are correct

**Comment by dyacob** [comment]
Thanks for reviewing @darcywong00 .  The words with periods are regular abbreviations that I think most users should recognize.  The words with Latin script, like "ንVOA", are references to external or international organizations.  Though not native Tigrinya words, but as the frequency counts indicate they get used often in Tigrinya text.  What is the best policy to follow here regarding inclusion/exclusion?

**Comment by darcywong00** [comment]
I guess it's ok to include a few. In general, I think we'd prefer the words match the specified language tag (`byn`).

File: release/gff/gff.byn.gff_blin/source/welcome.htm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    h2 { font: bold 12pt Tahoma; color: #4444cc; }
+  </style>
+</head>
+<body>
+
+<h1>Start Using GFF Amharic</h1>
+
+<p>
+    This lexical model is intended for use with the Blin Keyman Keyboard on mobile devices, see:
+    <a href="https://keyman.com/blin/">https://keyman.com/blin/</a>
```
Comment (darcywong00) [inline release/gff/gff.byn.gff_blin/source/welcome.htm:18]: Is this now?
```suggestion
    <a href="https://keyman.com/keyboards/gff_blin">https://keyman.com/keyboards/gff_blin</a>
```

**Comment by dyacob** [comment]
@darcywong00 a thought occurred to me on a potential problem.  The `gff_blin` keyboard uses the language ID `byn-Ethi` to distinguish itself from the `byn-Latn` convention (aka "Xaleget").  The lexicon uses the more general language ID `byn`; should this be changed to `byn-Ethi` as well in order for the keyboard to successfully find and download the lexicon?

**Comment by darcywong00** [comment]
> @darcywong00 a thought occurred to me on a potential problem. The `gff_blin` keyboard uses the language ID `byn-Ethi` to distinguish itself from the `byn-Latn` convention (aka "Xaleget"). The lexicon uses the more general language ID `byn`; should this be changed to `byn-Ethi` as well in order for the keyboard to successfully find and download the lexicon?

Yes, the mobile apps will expect an exact match of the language ID to associate the lexical-model with a keyboard. (casing doesn't matter).

**Comment by dyacob** [comment]
Thanks for confirming @darcywong00 , I've made the edit to sync the language ID.

**Comment by DavidLRowe** [comment]
@darcywong00 Does the .model_info file need to have the same language ID change?

File: release/gff/gff.byn.gff_blin/source/gff.byn.gff_blin.model.kps [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <FileType>.htm</FileType>
+    </File>
+  </Files>
+  <Keyboards/>
+  <LexicalModels>
+    <LexicalModel>
+      <Name>GFF Blin</Name>
+      <ID>gff.byn.gff_blin</ID>
+      <Languages>
+        <Language ID="byn-Ethi">Blin</Language>
```
Comment (darcywong00) [inline release/gff/gff.byn.gff_blin/source/gff.byn.gff_blin.model.kps:49]: Was this populated by Keyman Developer? We may want to add the script info like
```suggestion
        <Language ID="byn-Ethi">Blin (Ethiopic)</Language>
```


As @DavidLRowe pointed out, the .model_info should also have `byn-Ethi`.

File: release/gff/gff.byn.gff_blin/source/welcome.htm (hunk 1 above)
Comment (DavidLRowe) [reply release/gff/gff.byn.gff_blin/source/welcome.htm:18]: @dyacob Did you want to make this change? The original link doesn't work, so needs to be changed or removed.

**Comment by DavidLRowe** [comment]
@darcywong00 If the BCP 47 code changed to byn-Ethi, should the name of this model change to gff.byn-Ethi.gff_blin? I'm not totally clear on what is used to establish the connection between keyboard and model. Is it the BCP 47 code in the package? or does the middle part of the model name play a role. Or, turning the question around, is the BCP 47 code in the model name a convention that doesn't have to match exactly the code in the package? (I'll also note that in this case byn-Ethi and byn are equivalent tags.)

**Comment by dyacob** [comment]
@DavidLRowe , thanks for alerting me to the update needed, I thought I had gotten in a week earlier but it must have been in the wrong branch.

**Review by DavidLRowe — APPROVED** [review]

**Comment by dyacob** [comment]
@DavidLRowe & @darcywong00  -I'm very glad to see this PR merged and close.  I do have a troubleshooting question, I can post it elsewhere if you prefer.  I went into the Android Keyman app to install the Blin (byn-Ethi) dictionary, but it doesn't appear.  My steps:

1. Keyman > Installed languages > Bilin > Dictionary (Check for available dictionary)
2. The messages appear: "Checking for associated...." , "Downloading dictionary..." , "Dictionary download is finished" , "Resources successfully updated!"
3. The "Check for available dictionary" text remains, it is not replaced with the dictionary name.
4. Going into the Keyman app editor, or in local editors, and launching the Blin keyboard, the dictionary is not loaded, the top of the screen with the word options does not appear.
5. I've restarted Keyman, and restarted my phone, but no change. Other keyboards work fine.
6. Steps (1) and (2) are repeated, and the results are identical.

So I'm wondering if there is a bad metadata, or misaligned metadata, between the keyboard and dictionary.  It is interesting that the app can find, retrieve and install the dictionary, but it never becomes available when the keyboard is in use.

### Some version info:
* GFF Blin Keyboard v1.5.1
* Android Keyman: Version: 16.0.138

thanks!

**Comment by DavidLRowe** [comment]
Thanks for the good report. It might be better to raise an issue at https://github.com/keymanapp/keyman/issues

**Pull request merged**